DQM/BeamMonitor/plugins/BeamSpotProblemMonitor.cc

```diff
@@ -189,182 +150,156 @@ void BeamSpotProblemMonitor::analyze(const Event& iEvent,
+    }
 
-      //Reset it here for next lumi
-      BeamSpotError->Reset();
-      if(ALARM_ON_)BeamSpotError->Fill(BeamSpotFromDB_);
+    beamSpotStatusLumi_->ShiftFillLast( beamSpotStatus_, 0. , fitNLumi_ ); //beamSpotStatus_ =>0. (no collision, no tracks); =>1 (OK from scaler), =>-1 (No scalar results)
+    beamSpotStatusLumiAll_->setBinContent( currentlumi, beamSpotStatus_);
 
 
-    //Get quality report
-     MonitorElement* myQReport = dbe_->get(monitorName_+"FitFromScalars/BeamSpotError");
+  } else {
+    beamSpotStatusLumi_->ShiftFillLast( 0., 0., fitNLumi_ );
+  }//onlineMode_
 
-     const QReport * BeamSpotQReport = myQReport->getQReport("BeamSpotOnlineTest");  
+   //Reset it here for next lumi
+  beamSpotError_->Reset();
+  if(alarmOn_)beamSpotError_->Fill(beamSpotFromDB_);
 
-    if(BeamSpotQReport){  
-    		          /* S.Dutta : Commenting out these variables are not used and giving error with "-Werror=unused-variable" option
-    	                   float qtresult = BeamSpotQReport->getQTresult();
-                           int qtstatus   = BeamSpotQReport->getStatus() ; // get QT status value (see table below) */
-                      }
 
+  //Get quality report
+  const QReport * beamSpotQReport = beamSpotError_->getQReport("BeamSpotOnlineTest");
 
-   Ntracks_= 0;
+  if(beamSpotQReport){
+    /* S.Dutta : Commenting out these variables are not used and giving error with "-Werror=unused-variable" option
+    float qtresult = BeamSpotQReport->getQTresult();
+    int qtstatus   = BeamSpotQReport->getStatus() ; // get QT status value (see table below) */
+  }
 
+  nTracks_= 0;
 }
 
 //--------------------------------------------------------
 void BeamSpotProblemMonitor::endLuminosityBlock(const LuminosityBlock& lumiSeg,
-				     const EventSetup& iSetup) {
-  int nthlumi = lumiSeg.id().luminosityBlock();
+                                                const EventSetup& iSetup) {
+  const int nthlumi = lumiSeg.id().luminosityBlock();
   edm::LogInfo("BeamSpotProblemMonitor") << "endLuminosityBlock:: Lumi of the last event before endLuminosityBlock: " << nthlumi << endl;
-
-  if (onlineMode_ && nthlumi < nextlumi_) return;
-
 }
 //-------------------------------------------------------
 
 void BeamSpotProblemMonitor::endRun(const Run& r, const EventSetup& context){
 
-if(debug_)edm::LogInfo("BeamSpotProblemMonitor") << "endRun:: Clearing all the Maps "<<endl;
-     //Reset it end of job
-     BeamSpotError->Reset();
-}
-
-//--------------------------------------------------------
-void BeamSpotProblemMonitor::endJob(const LuminosityBlock& lumiSeg,
-			 const EventSetup& iSetup){
-  if (!onlineMode_) endLuminosityBlock(lumiSeg, iSetup);
-     //Reset it end of job
-     BeamSpotError->Reset();
+  if(debug_)edm::LogInfo("BeamSpotProblemMonitor") << "endRun:: Clearing all the Maps "<<endl;
+  //Reset it end of job
+  beamSpotError_->Reset();
```

Comment:
I'm not sure this `Reset` has any practical value since it will just get reset again at the first LuminosityBlock of the next Run.